sciLifeLab_utils/run_QC_analysis.py

```diff
@@ -47,7 +47,14 @@ def main(args):
         flowcells_dirs  = [os.path.join(sample_data_dir,flowcell) \
                 for flowcell in os.listdir(sample_data_dir) \
                 if os.path.isdir(os.path.join(sample_data_dir,flowcell))]
-
+        
+        # to adapt the diretory structure of X runs
+        checked_fc = [fc for fc in flowcells_dirs if re.match(r'^[A-Z]$', os.path.basename(fc))]
```

Comment:
Ok, I think I understand what is going on here... can you generalise the construction of flowcells_dirs in first place. Instead of using the for loop as I did in line 47 you can be more general using `glob`